mountpoint-s3-fs/src/superblock/path.rs

```diff
@@ -35,12 +35,19 @@ impl ValidKey {
             return Err(ValidKeyError::NotADirectory);
         };
 
-        let mut key = self.as_ref().to_owned();
-        let name_offset = key.len();
+        let name_offset = self.key.len();
+        // Allocate the new string with the correct capacity.
+        let mut key =
+            String::with_capacity(name_offset + name.len() + if kind == InodeKind::Directory { 1 } else { 0 });
+        key.push_str(&self.key);
         key.push_str(&name);
```

Comment:
> ensures that no slack capacity is wasted

regarding this aspect from the description, do you think slack capacity was added here, when pushing to the string? my assumption would be that initial allocation `self.as_ref().to_owned()` would use exactly the right amount of memory, so for `etag` and `storage_class` the saving is exactly one `usize` (previously used to track capacity)

Reply:
Yes, the slack capacity savings are only for keys. For `etag` and the children names we only save the `usize` field.